SharperUniverse/Core/BaseSharperSystem.cs

```diff
@@ -94,7 +94,7 @@ public bool EntityHasComponent(T component, SharperEntity entity)
 		    return Components.Find(c => c == component).Entity == entity;
 	    }
 
-	    public bool EntityHasComponent(BaseSharperComponent component, SharperEntity entity)
+	    public async Task<bool> EntityHasComponent(BaseSharperComponent component, SharperEntity entity)
```

Comment:
Uhhhhhhhhhhhhhhhhhhh? Elaborate please?

Reply:
You asked for it to be async in a previous review - I believe.

Reply:
hmmm really? Doesn't seem correct to me but if it needs to use TAP then I guess we can leave it, just seems odd.

Reply:
for context - https://github.com/RubyNova/SharperUniverse/pull/32#discussion_r240926590 - not sure if I misinterpreted or.